tests/org.jboss.tools.openshift.test/src/org/jboss/tools/openshift/test/ui/comparators/ProjectTreeSorterTest.java

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.test.ui.comparators;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+import org.eclipse.jface.viewers.ViewerSorter;
+import org.jboss.tools.openshift.internal.ui.comparators.ProjectViewerComparator;
+import org.jboss.tools.openshift.internal.ui.explorer.OpenShiftExplorerLabelProvider;
+import org.jboss.tools.openshift.internal.ui.treeitem.ObservableTreeItem;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.model.IBuildConfig;
+import com.openshift.restclient.model.IService;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ProjectTreeSorterTest {
+	
+	@Before
+	public void setUp() throws Exception {
+	}
+	
+	@Test
+	public void testServices() {
+		ViewerSorter comparator = ProjectViewerComparator.createProjectTreeSorter();
+		IService service1 = Mockito.mock(IService.class);
+		when(service1.getName()).thenReturn("s1");
+		when(service1.toString()).thenReturn("z");
+		ObservableTreeItem i1 = new ObservableTreeItem(service1);
+		IService service2 = Mockito.mock(IService.class);
+		when(service2.getName()).thenReturn("s2");
+		when(service2.toString()).thenReturn("a");
+		ObservableTreeItem i2 = new ObservableTreeItem(service2);
+		
+		assertTrue(comparator.compare(null, i1, i2) < 0);
```

Comment:
can you also check comparator.compare(null, i2, i1) and comparator.compare(null, i1, i1)?